### PR TITLE
Getting JSONAPI 1.0 responses

### PR DESCRIPTION
**Pull request by IAmJulianAcosta**
Hi changed some functions for getting a JSON API 1.0 valid response (including links and relationships). I really can't remember well what features specifically implemented in this PR, but I know that now the response is something like this:

``` javascript
{
  "data": [{
    "type": "articles",
    "id": "1",
    "attributes": {
      "title": "JSON API paints my bikeshed!"
    },
    "relationships": {
      "author": {
        "data": { "type": "people", "id": "9" }
      },
    },
    "links": {
      "self": "http://example.com/articles/1"
    }
  }],
  "included": [{
    "type": "people",
    "id": "9",
    "attributes": {
      "first-name": "Dan",
      "last-name": "Gebhardt",
      "twitter": "dgeb"
    },
    "links": {
      "self": "http://example.com/people/9"
    }
  }, {
    "type": "comments",
    "id": "5",
    "attributes": {
      "body": "First!"
    },
    "relationships": {
      "author": {
        "data": { "type": "people", "id": "2" }
      }
    },
    "links": {
      "self": "http://example.com/comments/5"
    }
  }, {
    "type": "comments",
    "id": "12",
    "attributes": {
      "body": "I like XML better"
    },
    "relationships": {
      "author": {
        "data": { "type": "people", "id": "9" }
      }
    },
    "links": {
      "self": "http://example.com/comments/12"
    }
  }]
}
```

### CODE REVIEW

**Comment by lsowen** [comment]
@egeriis Any thoughts on this PR?  I would like to use this package with ember-data, but the current response format for relationships isn't want is expected.

**Comment by IAmJulianAcosta** [comment]
@lsowen I created this because I'm using ember data, and currently is working. I have more things to commit (POST and PUT), but not finished yet

**Comment by egeriis** [comment]
Thank you for your contribution. It looks alright. There are a lot of unnecessary changes though—mostly added whitespace and other syntax issues. These needs to be fixed.

File: README.md [hunk 1]
```diff
@@ -49,13 +49,26 @@ use Request;
 
 class ApiController extends Controller
 {
-    public function handleRequest($modelName, $id = null)
+    /**
+     * Create handler name from request name
+     *
+     * @param $requestedResource string The name of the model (in plural)
```
Comment (mfn) [inline README.md]: Type type should come before the name, see also http://phpdoc.org/docs/latest/references/phpdoc/tags/param.html

File: src/EchoIt/JsonApi/Handler.php [hunk 2]
```diff
@@ -404,33 +405,28 @@ protected function handleSortRequest($cols, $model)
      * @return array
      * @throws Exception
      */
-    protected function parseRequestContent($content, $type)
-    {
-        $content = json_decode($content, true);
-        if (empty($content['data'])) {
+    protected function parseRequestContent( $content, $type ) {
+        $content = json_decode( $content, true );
+        $data    = $content['data'];
```
Comment (mfn) [inline src/EchoIt/JsonApi/Handler.php]: That's not good. You need to check if `data` is there in the first place; the original code did this properly.

**Comment by mfn** [comment]
I also tried to give my 2c, however
a) cosmetic code changes
b) introducing of whitespaces where none was
makes it harder to review than it should be.

On top, in the few seconds I looked I found a change which worsens the quality of the code; I did not look further as I take this as an alarm signal.

IMHO this attempt needs a fresh start ;-)

**Pull request closed without merging**